### DISCUSSION TITLE
Error: Couldn't parse Podfile.lock.

### DISCUSSION THREAD

**Discussion by pableiros** [swiftyfinch/Rugby#46] [Q&A] [answered]
I have a new project and when I run `rugby` I get the following messages:

```
Prepare Caclulate checksums ⏱
⛔️ Error: Couldn't parse Podfile.lock.
```

This is the content of the `Podfile.lock` file:

```
PODS:
  - Alamofire (5.4.3)
  - Fakery (5.1.0)
  - SwiftLint (0.43.1)
  - SwiftMessages (9.0.3):
    - SwiftMessages/App (= 9.0.3)
  - SwiftMessages/App (9.0.3)

DEPENDENCIES:
  - Alamofire
  - Fakery
  - SwiftLint
  - SwiftMessages

SPEC REPOS:
  trunk:
    - Alamofire
    - Fakery
    - SwiftLint
    - SwiftMessages

SPEC CHECKSUMS:
  Alamofire: e447a2774a40c996748296fa2c55112fdbbc42f9
  Fakery: a90caff00ca5cacde6c161c3eafc72314a03d34d
  SwiftLint: 99f82d07b837b942dd563c668de129a03fc3fb52
  SwiftMessages: 077f19126c24033fe24042237ecc20261adb46e4

PODFILE CHECKSUM: 4a9bedacd191201c2992a4d8627a1dc07dbe2e61

COCOAPODS: 1.10.2
````

`rugby` works perfectly on my old projects. Am I missing something to implement to make it work?

**Comment by swiftyfinch** [comment, marked as answer]
@pableiros Hello! Your `Podfile.lock` looks good.
Can you share your `Podfile`? I can create example and try to debug

**Reply by pableiros** [reply]
Hello!
This is the `Podfile` I'm using for the new project:

```ruby
platform :ios, '14.1'
use_frameworks!
inhibit_all_warnings!
install! 'cocoapods', :deterministic_uuids => false

target 'NewApp' do
    pod 'Alamofire'
    pod 'Fakery'
    pod 'SwiftLint'
    pod 'SwiftMessages'
end

post_install do |installer|
    installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
            config.build_settings['SWIFT_VERSION'] = '5'
            config.build_settings['GCC_WARN_INHIBIT_ALL_WARNINGS'] = "YES"
            config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '14.1'
            config.build_settings['SWIFT_VERSION'] = '5.3.3'
        end
    end
    installer.pods_project.build_configurations.each do |config|
        next unless config.name == 'Debug'
        config.build_settings['LD_RUNPATH_SEARCH_PATHS'] = [
          '$(FRAMEWORK_SEARCH_PATHS)'
        ]
    end
end
```

Thank you!

**Reply by swiftyfinch** [reply]
Could you delete `Podfile.lock` and try again to run `pod install` and `rugby`?

I made an example with your `Podfile` and it works in my environment.
But my `Podfile.lock` has different checksum:
```diff
diff Podfile.lock your_Podfile.lock 
28c28
< PODFILE CHECKSUM: edef4145838fe162cf7f2cd9f958b4fac54e65cc
---
> PODFILE CHECKSUM: 4a9bedacd191201c2992a4d8627a1dc07dbe2e61
```

Sorry, I don't have any other ideas for now.

**Reply by pableiros** [reply]
I deleted the `Podfile.lock`, change file permission, cloned again the repository and nothing.

Maybe is my environment. 

I will debug the `rugby` project and if I found a solution, I let you know.

**Reply by swiftyfinch** [reply]
Yeah, maybe `Rugby` can't find `Podfile.lock`.

Do you have a similar files structure?

<img width="216" alt="Screenshot 2021-08-25 at 17 42 53" src="https://user-images.githubusercontent.com/64660122/130811871-a8ce9e61-77b6-43cc-b57f-3f570a1d877d.png">

Also, you can attach the whole example if you don't have sensitive data.
Or send me an email

**Reply by pableiros** [reply]
I have the same file structure:

![Screen Shot 2021-08-25 at 13 05 17](https://user-images.githubusercontent.com/5769029/130850211-db107139-ad5c-4950-9ffa-b46d6ac0788b.png)

I downloaded and compiled the source code. I used the compile `rugby` version from `Xcode` and it worked!:

![Screen Shot 2021-08-25 at 11 29 11](https://user-images.githubusercontent.com/5769029/130850318-e1ed124e-230a-41a7-b81d-23e1ced3f954.png)

But when I use the `brew` version it doesn't work:

![Screen Shot 2021-08-25 at 13 04 06](https://user-images.githubusercontent.com/5769029/130850374-5a5c9de7-5c5f-45bb-810c-70512e034090.png)

So I notice I had the `1.6.0` version installed from `brew`.

I deleted that version to install the last version from `mint` and now it works !!

![Screen Shot 2021-08-25 at 13 16 16](https://user-images.githubusercontent.com/5769029/130851620-ee862dfb-8b7f-4f9f-b5d8-ac1de9696a37.png)

Sorry for all the throubles and thank you for your time and for your awesome tool.

**Reply by swiftyfinch** [reply]
Oh, I see. Thank you for the detailed investigation 👍🏻
Yeah, the `Brew` version deprecated, because now 🏈 Rugby is open source.

The [Mint](https://github.com/yonaskolb/Mint) is a more convenient tool 🌱
You can install any version or `master` revision.
